libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreInspectorViewController.m

```diff
@@ -88,14 +88,28 @@ + (SFSmartStoreInspectorViewController *) sharedInstance
 
 #pragma mark - Present / dimiss
 
-+ (void) present
++ (void) present:(UIViewController*)currentViewController
 {
-    [[SFRootViewManager sharedManager] pushViewController:[SFSmartStoreInspectorViewController sharedInstance]];
+    if (![NSThread isMainThread]) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [SFSmartStoreInspectorViewController present:currentViewController];
+        });
+        return;
+    }
+
+    [currentViewController presentViewController:[SFSmartStoreInspectorViewController sharedInstance] animated:NO completion:nil];
```

Comment:
It's more than a one liner if you are not running on the main thread.
Also we have already documented present/dismiss so I didn't want to drop them entirely (even though I changed the signature of present).
But anyway, they don't have to use those methods, they can get a controller and present it themselves.